openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -0,0 +1,240 @@
+"""
+Declaration of CourseOverview model
+"""
+
+from datetime import datetime as date_time  # the as clause avoids naming conflicts the DateTime field
+import json
+from util.date_utils import strftime_localized
+
+import django.db.models
+from django.db.models.fields import *
+from django.utils.timezone import UTC
+from django.utils.translation import ugettext
+
+from certificates.api import get_active_web_certificate
+from xmodule import course_metadata_utils
+from courseware.courses import course_image_url
+from xmodule.modulestore.django import modulestore
+from xmodule_django.models import CourseKeyField, UsageKeyField
+
+class CourseOverview(django.db.models.Model):
+    """Model for storing and caching basic information about a course.
+
+    This model contains basic course metadata such as an ID, display name,
+    image URL, and any other information that would be necessary display
+    a course as part of a user dashboard or enrollment API.
+    """
+
+    # Course identification
+    id = CourseKeyField(db_index=True, primary_key=True, max_length=255)
+    _location = UsageKeyField(max_length=255)
+    display_name = TextField(null=True)
+    display_number_with_default = TextField()
+    display_org_with_default = TextField()
+
+    # Start/end dates
+    start = DateTimeField(null=True)
+    end = DateTimeField(null=True)
+    advertised_start = TextField(null=True)
+
+    # URLs
+    course_image_url = TextField()
+    facebook_url = TextField(null=True)
+    social_sharing_url = TextField(null=True)
+
+    # Certification data
+    certificates_display_behavior = TextField(null=True)
+    certificates_show_before_end = BooleanField()
+    has_active_web_certificates = BooleanField()
+    cert_name_short = TextField()
+    cert_name_long = TextField()
+    lowest_passing_grade = DecimalField(max_digits=5, decimal_places=2)
```

Comment:
The reason I put it there (along with end_of_course_survey_url) was because they were both used only in calculating whether a certificate should be shown or not. However, I guess that's more of an implementation detail, so I'll put it in it's own section